duckshoot.py

```diff
@@ -0,0 +1,337 @@
+#!/usr/bin/env python
+# coding: Latin
+
+import json
+from my_button import MyScale
+# Load all standard tools for image processing challenges
+from img_base_class import *
+
+
+# Image stream processing thread
+class StreamProcessor(threading.Thread):
+    def __init__(self, screen=None, camera=None, drive=None, colour="any"):
+        super(StreamProcessor, self).__init__()
+        self.camera = camera
+        image_width, image_height = self.camera.resolution
+        self.drive = drive
+        self.drive.should_normalise_motor_speed = False
+        self.screen = screen
+        self.stream = picamera.array.PiRGBArray(camera)
+        self.event = threading.Event()
+        self.terminated = False
+        self.MAX_AREA = 2000  # Largest target to shoot at
+        self.MIN_CONTOUR_AREA = 400
+        self.MAX_TARGET_SIZE = 2000
+        self.MAX_TARGET_WIDTH = 80
+        self.AIM_OFFSET = 45
+        self._colour = colour
+        self.found = False
+        self.cycle = 0
+        self.target_number = 0
+        self.menu = False
+        self.last_t_error = 0
+        self.TURN_P = 0.8
+        self.TURN_D = 0.4
+        self.CROP_WIDTH = 320
+        self.CROP_HEIGHT = 60
+        self.image_centre_x = self.CROP_WIDTH / 2.0
+        self.image_centre_y = self.CROP_HEIGHT / 2.0
+        self.CROP_H_OFFSET = 160
+        self.CROP_V_OFFSET = 180
+        self.colour_bounds = json.load(open('duckshoot.json'))
+        self.hsv_lower = (0, 0, 0)
+        self.hsv_upper = (0, 0, 0)
+        self.DRIVING = True
+        self.tracking = False
+        # Why the one second sleep?
```

Comment:
no, really? why the one sec sleep? if we're happy with it, lets remove this comment.

Reply:
I haven't tested setting it smaller yet. its recommended to 'leave the camera to stabilise for a while' but I don't know if that's necessary for us, since we're fixing some of the auto evaluated settings. I'm not happy with the delay, just haven't got round to testing what happens if its removed.

Reply:
tested, seems unnecessary. I've removed it and the 2second sleep in the DuckShoot class run method and ti still runs without error.